include/odp/api/spec/ipsec.h

```diff
@@ -1206,6 +1210,14 @@ typedef struct odp_ipsec_status_t {
  * restored. The amount and content of packet data before the IP header is
  * undefined.
  *
+ * Additional TFC padding might be present after packet contents. For ESP
+ * transport mode ODP does not truncate such padding, it up to an application
+ * to detect and drop it. For ESP tunnel mode, received side can use total
+ * (IPv4) or payload (IPv6) length from internal headers to drop such TFC
+ * padding. If tfc_padding_truncate capability is set, implementation will
+ * truncate received packets automatically. Otherwise ODP application has to
+ * truncate packets manually.
+ *
```

Comment:
I suggest rewording like this (after dropping tfc_padding_truncate capa):

An inbound ESP packet may contain TFC padding. odp_ipsec_in() might or might not remove the padding. When TFC padding is not removed, the resulting ODP packet will extend beyond the end of the IP packet it contains or the resulting IP packet will extend beyond the end of the IP payload protocol. An ODP application can detect and remove such padding by inspecting the length fields of the relevant protocol headers in the result packet.

Reply:
We added this capability mainly coz NXP were removing the padding and Cavium was not removing the padding. If from the application point of view you are fine with the wording you have suggested above then this capability can be removed.

Reply:
This is useful information for the application to know about how the implementation will behave. Applications are free to ignore this if they wish, but I see no harm in keeping it.

Reply:
> I see no harm in keeping it

I understand you refer to the tfc_padding_truncate capability. Note that the proposal to get rid of it and discussion on that (with comments from Petri too) are located elsewhere in this tool.

That said, the harm of keeping the capability (in addition to growing the API) is that the capability makes a promise that may (depending on what value zero means) unnecessarily restrict implementations where the fate of the padding is not the same globally in the whole system and at all times (e.g. NIC based IPsec offload where padding is removed or not depending no through which NIC the packet came in, or a system where removal of the padding depends somehow on the packet).

If the capability is kept, its meaning needs to be documented better, including specifying what kinds of TFC padding it applies to (see other comments regarding that).

Reply:
OK. Agree with this approach for Tiger Moth.

Reply:
The reference to IKEv2 ESP_TFC_PADDING_NOT_SUPPORTED notification should be removed. IKE protocol details do not belong to this API spec which should just describe the API. One can read the IKE RFC for IKE details or one might not even use IKE but something else.

Even when IKE is used, the remark is inaccurate. Even if we send the notification for some SA, it does not guarantee that the peer does not send us TFC padding but just declares that we do not accept such packets if we receive them.